### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by mbevc1** [weaveworks/eksctl#6287]
### What were you trying to accomplish?

Deleting EKS cluster provisioned with addons.

### What happened?

When deleting the cluster first step is cordoning the nodes. EBS addon is running controller replica and is blocking cordoning of the nodes. We should probably be removing the addons first to ensure only workloads are running before cordoning.

### How to reproduce it?

Provision EKS cluster with a addon (EBS). Then delete is using `eksctl delete cluster -f config.yaml`

**Logs**

```
2023-02-13 14:36:56 [ℹ]  deleting EKS cluster "mb"
2023-02-13 14:36:57 [ℹ]  will drain 0 unmanaged nodegroup(s) in cluster "mb"
2023-02-13 14:36:57 [ℹ]  starting parallel draining, max in-flight of 1
2023-02-13 14:36:58 [ℹ]  cordon node "ip-192-168-29-32.eu-west-1.compute.internal"
2023-02-13 14:36:58 [ℹ]  cordon node "ip-192-168-46-245.eu-west-1.compute.internal"
2023-02-13 14:36:58 [ℹ]  cordon node "ip-192-168-92-221.eu-west-1.compute.internal"
2023-02-13 14:38:12 [!]  1 pods are unevictable from node ip-192-168-29-32.eu-west-1.compute.internal
2023-02-13 14:39:14 [!]  1 pods are unevictable from node ip-192-168-29-32.eu-west-1.compute.internal
2023-02-13 14:40:16 [!]  1 pods are unevictable from node ip-192-168-29-32.eu-west-1.compute.internal
2023-02-13 14:41:18 [!]  1 pods are unevictable from node ip-192-168-29-32.eu-west-1.compute.internal
2023-02-13 14:42:20 [!]  1 pods are unevictable from node ip-192-168-29-32.eu-west-1.compute.internal
```

**Anything else we need to know?**

**Versions**

```
$ eksctl info
```
eksctl version: 0.129.0
kubectl version: v1.26.1
OS: linux

**Comment by TiberiuGC** [comment]
Hi @mbevc1 ! As per our documentation on how to delete clusters [here](https://eksctl.io/usage/creating-and-managing-clusters/), Pod Disruption Budget policies are preventing the EBS addon from being properly removed. You should run your command with `--disable-nodegroup-eviction` flag. i.e.

`eksctl delete cluster -f cluster.yaml --disable-nodegroup-eviction`

**Comment by mbevc1** [comment]
Thanks @TiberiuGC for pointing ou this doc note. I guess removing the addon first or in parallel could prevent this issue?

**Comment by TiberiuGC** [comment]
I think you're right, removing the addon first would probably work as-well.

**Comment by mbevc1** [comment]
Any more decisions on this?

**Comment by mbevc1** [comment]
`keep-alive` :)

**Comment by dabcoder** [comment]
> I think you're right, removing the addon first would probably work as-well.

Just running into this issue as we speak, so I'd suggest adding a note about removing the add in on this page: https://eksctl.io/usage/creating-and-managing-clusters/

**Comment by mbevc1** [comment]
`keep-alive` again :)

**Comment by AmitBenAmi** [comment]
@mbevc1 have you tried to run with `--disable-nodegroup-eviction`?

**Comment by mbevc1** [comment]
I haven't tried that one, but looks like it could work. But I reckon we still want to have a graceful and build-in support to deprovision add-ons

**Comment by EmpeRoar** [comment]
> Hi @mbevc1 ! As per our documentation on how to delete clusters [here](https://eksctl.io/usage/creating-and-managing-clusters/), Pod Disruption Budget policies are preventing the EBS addon from being properly removed. You should run your command with `--disable-nodegroup-eviction` flag. i.e.
> 
> `eksctl delete cluster -f cluster.yaml --disable-nodegroup-eviction`

this did not work for me. im still having the issue

**Comment by mbevc1** [comment]
`keep-alive`

**Comment by m0un10** [comment]
It's probably worth calling out that some addons get added by default for new clusters. So, I don't think there are actually any cases now where the `delete` will actually work. At least `--disable-nodegroup-eviction` unblocks though! :)

**Comment by yuxiang-zhang** [comment]
> It's probably worth calling out that some addons get added by default for new clusters.

Indeed, the coredns addon now has a [default Pod Disruption Budget (PDB)](https://aws.amazon.com/blogs/containers/recent-changes-to-the-coredns-add-on/#:~:text=Issue%20%231679%20requested%20adding%20a%20Pod%20Disruption%20Budget%20(PDB)%20to%20the%20coredns%20addon%20and%20was%20implemented%20in%20v1.9.3%2Deksbuid.5%20and%20higher.), preventing the pod to be evicted when draining the nodegroups. As a workaround for now, we can use `eksctl delete cluster --disable-nodegroup-eviction` or `eksctl delete nodegroup --disable-eviction`.

**Comment by dabcoder** [comment]
Just thinking out loud but would it make sense (would it be feasible?) to add a command flag to remove add-ons? e.g. `eksctl delete cluster --remove-add-ons`

**Comment by N-LamTruong** [comment]
> > It's probably worth calling out that some addons get added by default for new clusters.
> 
> Indeed, the coredns addon now has a [default Pod Disruption Budget (PDB)](https://aws.amazon.com/blogs/containers/recent-changes-to-the-coredns-add-on/#:~:text=Issue%20%231679%20requested%20adding%20a%20Pod%20Disruption%20Budget%20(PDB)%20to%20the%20coredns%20addon%20and%20was%20implemented%20in%20v1.9.3%2Deksbuid.5%20and%20higher.), preventing the pod to be evicted when draining the nodegroups. As a workaround for now, we can use `eksctl delete cluster --disable-nodegroup-eviction` or `eksctl delete nodegroup --disable-eviction`.

It works for me! Thanks bro
